core/src/main/java/org/apache/brooklyn/core/entity/AbstractEntity.java

```diff
@@ -1209,59 +1208,18 @@ public void remove(AttributeSensor<?> attribute) {
     public class BasicConfigurationSupport extends AbstractConfigurationSupportInternal {
 
         @Override
-        public <T> T get(ConfigKey<T> key) {
-            return configsInternal.getConfig(key);
-        }
-
-        @Override
-        public <T> T set(ConfigKey<T> key, T val) {
+        protected <T> void assertValid(ConfigKey<T> key, T val) {
             ConfigConstraints.assertValid(AbstractEntity.this, key, val);
-            return setConfigInternal(key, val);
-        }
-
-        @Override
-        public <T> T set(ConfigKey<T> key, Task<T> val) {
-            return setConfigInternal(key, val);
-        }
-
-        @Override
-        public ConfigBag getBag() {
-            return configsInternal.getAllConfigBag();
-        }
-
-        @Override
-        public ConfigBag getLocalBag() {
-            return configsInternal.getLocalConfigBag();
-        }
-
-        @Override
-        public Maybe<Object> getRaw(ConfigKey<?> key) {
-            return configsInternal.getConfigRaw(key, true);
         }
-
-        @Override
-        public Maybe<Object> getLocalRaw(ConfigKey<?> key) {
-            return configsInternal.getConfigRaw(key, false);
-        }
-
-        @Override
-        public void addToLocalBag(Map<String, ?> vals) {
-            configsInternal.addToLocalBag(vals);
-        }
-
-        @Override
-        public void removeFromLocalBag(String key) {
-            configsInternal.removeFromLocalBag(key);
+        
+        protected AbstractConfigMapImpl<?> getConfigsInternal() {
```

Comment:
I like that pattern most places.  However if it is only protected usage and the return type is in an internal package I don't think there is any serious risk of confusion and it is not worth the burden of internal interfaces.  (While not too hard to write it does make navigating the code much harder.)